cmd/main.go

```diff
@@ -110,10 +111,28 @@ func main() {
 	nodeName := os.Getenv("NODE_NAME")
 	node, err := k8sClientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
 	if err != nil {
-		logger.Info("Error retrieving the Node from the index for a given node. Error :", zap.Error(err))
 		if errors.IsNotFound(err) {
 			runtimeu.HandleError(fmt.Errorf("node '%s' no longer exist in the cluster", nodeName))
 		}
+		// Do multiple retries if there is connection error.
+		logger.Info("Retrying connection to node")
+		errRetry := nodeupdater.ErrorRetry(logger, func() (error, bool) {
+			node, err = k8sClientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			if err != nil {
+				var shouldStop bool
+				errString := err.Error()
+				if strings.Contains(errString, "tcp") {
+					shouldStop = false
+				} else {
+					shouldStop = true
+				}
+				return err, shouldStop // Skip retry if its not connection error
+			}
+			return nil, true
+		})
+		if errRetry != nil {
+			logger.Fatal("Error retrieving the Node from the index for a given node. Error :", zap.Error(err))
```

Comment:
can we log following error instead of this

`Failed to get node details. Error: <error>`